### PR TITLE
BSM From RSU to ODE Ingest and Decoding

### PR DESCRIPTION
**Pull request by drewjj**
# PR Details
## Description

Includes the changes made to the jpo-ode-svcs to re-implement UDP receiving for BSM messages over port 46800. This allows the machine the ODE is running on to receive BSM messages automatically forwarded to it from an RSU or another machine and be received by the ODE to then decode and process the BSM as if it had been uploaded in any other way.

Most of the UDP code is revived code from an earlier UDP implementation but brought up to date with the current workings of the ODE and decoded with a new set of scripts that Leidos has provided on the main USDOT repository for the jpo-ode.

New wiki page for this new feature: [BSM from RSU to DOE UDP](https://github.com/usdot-jpo-ode/jpo-ode/wiki/RSU-to-ODE-over-UDP-ingest-and-decoding)

## Related Issue

#403 

## Motivation and Context

#403 

## How Has This Been Tested?

This was tested locally by setting up the ODE on a Linux machine with open ports for 46800. An RSU was then configured to automatically forward BSM messages to the machine on port 46800 over IPv4. Although it was tested with IPv4, the UDP support will more than likely work with IPv6 if the docker-compose.yml is setup to make the docker containers support IPv6 connections. (They do not at the moment)

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -33,7 +33,8 @@ services:
       - ${DOCKER_SHARED_VOLUME_WINDOWS}/var/run/docker.sock:/var/run/docker.sock
 
   ode:
-    image: usdotjpoode/jpo-ode:1.0
+    build: .
```
Comment (dan-du-car) [inline docker-compose.yml:36]: Everything looks good. A minor question whether building the image from source or pull from this dockerhub: https://hub.docker.com/u/usdotjpoode?

Comment (drewjj) [reply docker-compose.yml:36]: Building from source ensures it is using all the code changes that have been made within the ODE portion of the project although building from the image is good if it is the latest and the new changes have been published if I am not mistaken.

File: docker-compose.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ services:
       KAFKA_ADVERTISED_HOST_NAME: ${DOCKER_HOST_IP}
       KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
       KAFKA_AUTO_CREATE_TOPICS_ENABLE: "true"
-      KAFKA_CREATE_TOPICS: "topic.OdeBsmPojo:1:1,topic.OdeBsmJson:1:1,topic.FilteredOdeBsmJson:1:1,topic.OdeTimJson:1:1,topic.OdeTimBroadcastJson:1:1,topic.J2735TimBroadcastJson:1:1,topic.OdeDriverAlertJson:1:1,topic.Asn1DecoderInput:1:1,topic.Asn1DecoderOutput:1:1,topic.Asn1EncoderInput:1:1,topic.Asn1EncoderOutput:1:1,topic.SDWDepositorInput:1:1"
+      KAFKA_CREATE_TOPICS: "topic.OdeBsmPojo:1:1,topic.OdeBsmJson:1:1,topic.FilteredOdeBsmJson:1:1,topic.OdeTimJson:1:1,topic.OdeTimBroadcastJson:1:1,topic.J2735TimBroadcastJson:1:1,topic.OdeDriverAlertJson:1:1,topic.Asn1DecoderInput:1:1,topic.Asn1DecoderOutput:1:1,topic.Asn1EncoderInput:1:1,topic.Asn1EncoderOutput:1:1,topic.SDWDepositorInput:1:1,topic.OdeTIMCertExpirationTimeJson:1:1"
```
Comment (dan-du-car) [inline docker-compose.yml:20]: Also adding "topic.OdeRawEncodedMessageJson:1:1"

**Comment by drewjj** [comment]
I added the topic.OdeRawEncodedMessageJson:1:1 to the docker-compose.yml

**Review by dan-du-car — APPROVED** [review]
Approve for now. Will need to update if we decide to leverage dockerhub later.

**Pull request merged**